dev/plugins/hu.elte.txtuml.export.plantuml/src/hu/elte/txtuml/export/plantuml/generator/PlantUmlGenerator.java

```diff
@@ -0,0 +1,103 @@
+package hu.elte.txtuml.export.plantuml.generator;
+
+import java.io.ByteArrayInputStream;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTParser;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.Type;
+
+import hu.elte.txtuml.export.plantuml.exceptions.ExportRuntimeException;
+import hu.elte.txtuml.export.plantuml.exceptions.PreCompilationError;
+import hu.elte.txtuml.export.plantuml.exceptions.SequenceDiagramStructuralException;
+
+/**
+ * 
+ * This Class combines the compiler and the preCompiler and runs them. Provides
+ * the required utility functions for them
+ * 
+ * @author Zoli
+ *
+ */
+public class PlantUmlGenerator {
+
+	private IFile targetFile;
+	private CompilationUnit sourceCU;
+
+	private PlantUmlPreCompiler preCompiler;
+	private PlantUmlCompiler compiler;
+
+	public PlantUmlGenerator(IFile targetFile, CompilationUnit source) {
+
+		this.targetFile = targetFile;
+		this.sourceCU = source;
+	}
+
+	public void generate() throws SequenceDiagramStructuralException, PreCompilationError {
+
+		preCompiler = new PlantUmlPreCompiler();
+		sourceCU.accept(preCompiler);
+		if (!preCompiler.getErrors().isEmpty()) {
+
+			String messages = "";
+
+			for (Exception ex : preCompiler.getErrors()) {
+				messages += "\n" + ex.getMessage();
+			}
+			throw new PreCompilationError(messages);
+		}
+
+		Type superClass = preCompiler.getSuperClass();
+		while (superClass != null) {
+			CompilationUnit cu = null;
+
+			if (superClass != null) {
```

Comment:
Is this if statement needed, since we check the same condition in the loop condition?

Reply:
It is not needed.